lib/esp32.toit

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2021 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+ESP_RST_UNKNOWN   ::= 0 // Reset reason can not be determined.
+ESP_RST_POWERON   ::= 1 // Reset due to power-on event.
+ESP_RST_EXT       ::= 2 // Reset by external pin (not applicable for ESP32).
+ESP_RST_SW        ::= 3 // Software reset via esp_restart.
+ESP_RST_PANIC     ::= 4 // Software reset due to exception/panic.
+ESP_RST_INT_WDT   ::= 5 // Reset (software or hardware) due to interrupt watchdog.
+ESP_RST_TASK_WDT  ::= 6 // Reset due to task watchdog.
+ESP_RST_WDT       ::= 7 // Reset due to other watchdogs.
+ESP_RST_DEEPSLEEP ::= 8 // Reset after exiting deep sleep mode.
+ESP_RST_BROWNOUT  ::= 9 // Brownout reset (software or hardware).
+ESP_RST_SDIO      ::= 10 // Reset over SDIO.
+
+// Enum constants from esp_sleep_source_t in esp-idf/components/esp32/include/esp_sleep.h.
+WAKEUP_UNDEFINED ::= 0 // In case of deep sleep, reset was not caused by exit from deep sleep.
+WAKEUP_ALL       ::= 1 // Not a wakeup cause, used to disable all wakeup sources with esp_sleep_disable_wakeup_source.
+WAKEUP_EXT0      ::= 2 // Wakeup caused by external signal using RTC_IO.
+WAKEUP_EXT1      ::= 3 // Wakeup caused by external signal using RTC_CNTL.
+WAKEUP_TIMER     ::= 4 // Wakeup caused by timer.
+WAKEUP_TOUCHPAD  ::= 5 // Wakeup caused by touchpad.
+WAKEUP_ULP       ::= 6 // Wakeup caused by ULP program.
+WAKEUP_GPIO      ::= 7 // Wakeup caused by GPIO (light sleep only).
+WAKEUP_UART      ::= 8 // Wakeup caused by UART (light sleep only).
+
+/**
+  Enters deep sleep for the specified duration (up to 24h) and does not return. Exiting
+  deep sleep causes the ESP32 to start over from main.
+*/
+deep_sleep duration/Duration -> none:
+  __deep_sleep__ duration.in_ms
+
+/**
+  Returns one of the ESP_RST_* enum values that indicate why the ESP32 was reset.
+*/
+reset_reason -> int:
+  #primitive.esp32.reset_reason
+
+/**
+  Returns one of the WAKEUP_* enum values that indicate why the ESP32 was woken up.
+*/
+wakeup_cause -> int:
+  #primitive.esp32.wakeup_cause
+
+/**
+  Returns the total number of microseconds this device has been running (including deep sleep).
+  NOTE: currently boot time of FreeRTOS is not included (this might be significant).
```

Comment:
```suggestion
Returns the total number of microseconds this device has been running (including deep sleep).
NOTE: currently boot time of FreeRTOS is not included (this might be significant).
```